packages: remove remnants of kernels below 6.6

namiltd opened the pull request:
Remove remnants of kernels below 6.6.

hauke commented on package/kernel/linux/modules/netfilter.mk:
```diff
@@ -823,7 +823,7 @@ define KernelPackage/ipt-clusterip
   KCONFIG:=$(KCONFIG_IPT_CLUSTERIP)
   FILES:=$(foreach mod,$(IPT_CLUSTERIP-m),$(LINUX_DIR)/net/$(mod).ko)
   AUTOLOAD:=$(call AutoProbe,$(notdir $(IPT_CLUSTERIP-m)))
-  $(call AddDepends/ipt,+kmod-nf-conntrack @LINUX_6_1)
```
I think this package is only activated on Linux 6.1. Probably remove it.

namiltd replied:
Done

neheb commented:
ath79 needs refreshing because of ar8216.c changes.

I actually wonder about this patch: https://github.com/openwrt/openwrt/blob/main/target/linux/ath79/patches-6.6/730-ar8216-make-reg-access-atomic.patch

3 platforms set CONFIG_AR8216_PHY=y , ath79, ramips, and lantiq.

ath79 apparently doesn't use interrupts. ramips from a quick glance no. Not sure about lantiq.

Is the comment still valid? ping @981213 @Ansuel

neheb commented:
maybe not? https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/drivers/net/dsa/lantiq_gswip.c?h=v6.11#n836

namiltd commented:
The compilation errors are due to https://github.com/openwrt/openwrt/commit/3660ddb8ab6e6e4673576a0ee520ea38c67b9d75 . They are not related to removing the remnants of 6.1 so I think we should create a separate PR for them

namiltd committed:
packages: remove remnants of kernels below 6.6

Remove remnants of kernels below 6.6.

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>
Link: https://github.com/openwrt/openwrt/pull/16432
Signed-off-by: Robert Marko <robimarko@gmail.com>

hauke commented:
I refreshed the ath79 patches. 

The comment in lantiq_gswip.c is about a hardware feature. The switch hardware is able to poll the PHYs automatically in hardware and then adapt the settings in the MAC. This is conflicting with Linux when Linux polls the PHYs. The HW expects that the PHYs are working according to the generic base IEEE specification and can not handle bugs or new versions of the specification. The Linux drivers are handing many hardware bugs, we ran into such cases in OpenWrt.

robimarko commented on include/netfilter.mk:
```diff
@@ -131,9 +131,6 @@ $(eval $(call nf_add,IPT_IPOPT,CONFIG_NETFILTER_XT_TARGET_HL, $(P_XT)xt_HL))
 $(eval $(call nf_add,IPT_IPRANGE,CONFIG_NETFILTER_XT_MATCH_IPRANGE, $(P_XT)xt_iprange))
 
-#clusterip
-$(eval $(call nf_add,IPT_CLUSTERIP,CONFIG_IP_NF_TARGET_CLUSTERIP, $(P_V4)ipt_CLUSTERIP))
```
`CONFIG_IP_NF_TARGET_CLUSTERIP` still exists in 6.6, why are you removing it?

robimarko replied:
Ahh, I see, the package using it is dropped later.

robimarko commented:
Thanks! Rebased on top of main and merged!

namiltd commented:
One flaw: https://github.com/openwrt/openwrt/pull/16450
